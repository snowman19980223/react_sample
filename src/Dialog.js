import React from 'react'
import {Dialog as ReachDialog} from '@reach/dialog'
import raw from 'raw.macro'
import styled, {createGlobalStyle} from 'styled-components'
import PropTypes from 'prop-types'
import {space, color} from 'styled-system'
import systemPropTypes from '@styled-system/prop-types'
import {X} from '@primer/octicons-react'
import StyledOcticon from './StyledOcticon'
import {LAYOUT} from './constants'
import theme from './theme'
import Text from './Text'
import Flex from './Flex'

const reachStyles = raw('@reach/dialog/styles.css')

const ReachGlobalStyle = createGlobalStyle`
  ${reachStyles}

  [data-reach-dialog-overlay] {
    z-index: 1000002; /* Higher than the Dropdown and Tooltip */
  }
`

export const StyledDialog = styled(ReachDialog)`
  box-shadow: 0px 4px 32px rgba(0, 0, 0, 0.35);
  border-radius: 4px;
  padding: 0 !important;
  position: relative;

  @media screen and (max-width: 750px) {
    width: 100vw !important;
    margin: 0 !important;
    border-radius: 0;
    height: 100vh;
  }

  ${LAYOUT}
  ${space}
  ${color}
`

const UnstyledButton = styled(Flex).attrs({
  as: 'button'
})`
  background: none;
  border: none;
  padding: 0;

  position: absolute;
  top: 16px;
  right: 16px;
`

const DialogHeaderBase = styled(Flex)`
  border-radius: 4px 4px 0px 0px;
  border-bottom: 1px solid #dad5da;

  @media screen and (max-width: 750px) {
    border-radius: 0px;
  }
`

function DialogHeader({theme, children, ...rest}) {
  if (React.Children.toArray(children).every(ch => typeof ch === 'string')) {
    children = (
      <Text theme={theme} color="gray.9" fontSize={1} fontWeight="bold" fontFamily="sans-serif">
        {children}
      </Text>
    )
  }

  return (
    <DialogHeaderBase theme={theme} p={3} {...rest}>
      {children}
    </DialogHeaderBase>
  )
}

const Dialog = ({children, ...props}) => {
  return (
    <>
      <StyledDialog {...props}>
        <UnstyledButton onClick={props.onDismiss}>
          <StyledOcticon icon={X} />
        </UnstyledButton>
        {children}
      </StyledDialog>
      <ReachGlobalStyle />
    </>
  )
}

Dialog.defaultProps = {theme}

Dialog.propTypes = {
  ...LAYOUT.propTypes,
  ...systemPropTypes.space,
  ...systemPropTypes.color,
  children: PropTypes.node.isRequired,
  isOpen: PropTypes.bool.isRequired,
  onDismiss: PropTypes.func.isRequired,
  theme: PropTypes.object
}

DialogHeader.defaultProps = {
  backgroundColor: 'gray.1',
  theme
}

DialogHeader.propTypes = {
  ...Flex.propTypes
}

Dialog.Header = DialogHeader
export default Dialog
