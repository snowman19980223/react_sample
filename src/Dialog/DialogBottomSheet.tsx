import React, {useEffect, useRef, useState, PropsWithChildren, MouseEvent, TouchEvent, ChangeEvent} from 'react'
import styled from 'styled-components'
import {SxProp} from '../sx'
import {useRefObjectAsForwardedRef} from '../hooks/useRefObjectAsForwardedRef'
import Box from '../Box'
import {get} from '../constants'
import {useMedia} from '../hooks/useMedia'

export const ANIMATION_DURATION = 300
export const FULL_HEIGHT = 95
export const HALF_HEIGHT = 50

/**
 * Props to customize the rendering of the Dialog.
 */
export interface DialogBottomSheetProps extends SxProp {
  /**
   * This method is invoked when a gesture to close the dialog is used
   */
  onClose: () => void

  /**
   * Content that appears in the draggable header area.
   */
  header: React.ReactNode

  /**
   * Default: "dialog". The ARIA role to assign to this dialog.
   * @see https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal
   * @see https://www.w3.org/TR/wai-aria-practices-1.1/#alertdialog
   */
  role?: 'dialog' | 'alertdialog'

  /**
   * Identifies the element (or elements) that labels the element it is applied to.
   */
  ariaLabelledby: string

  /**
   * Identifies the element (or elements) that describes the element on which the attribute is set.
   */
  ariaDescribedby: string

  /**
   * Indicates whether an element is modal when displayed.
   */
  ariaModal: boolean
}

export default React.forwardRef<HTMLDivElement, PropsWithChildren<DialogBottomSheetProps>>((props, forwardedRef) => {
  const {onClose, children, role, ariaLabelledby, ariaDescribedby, ariaModal, header, sx} = props

  // üîÑ STATES

  const [open, setIsOpen] = useState<boolean>(false)
  const [snappedHeight, setSnappedHeight] = useState<number>(HALF_HEIGHT)
  const [fireDelayedOnClose, setFireDelayedOnClose] = useState<boolean>(false)

  // üìé REFERENCES

  const dialogRef = useRef<HTMLDivElement>(null)
  const startY = useRef(0)
  const startHeight = useRef(0)
  const isDragging = useRef(false)

  useRefObjectAsForwardedRef<HTMLDivElement>(forwardedRef, dialogRef)

  // üßë‚Äçü¶Ω ACCESSIBILITY

  const isReduced = useMedia('(prefers-reduced-motion: no-preference)', false) === false

  // ü•ä ACTIONS

  const showBottomSheet = () => {
    setIsOpen(true)
    updateSheetHeight(HALF_HEIGHT)
  }

  const hideBottomSheet = async () => {
    setIsOpen(false)
    if (isReduced) {
      onClose()
    } else {
      setFireDelayedOnClose(true)
    }
  }

  const updateSheetHeight = (height: number) => {
    if (!dialogRef.current) return
    dialogRef.current.style.height = `${height}dvh`
    setSnappedHeight(height)
  }

  // üé™ EVENTS

  const dragStop = (e: MouseEvent | TouchEvent) => {
    if (!dialogRef.current) return

    e.preventDefault()

    isDragging.current = false
    const sheetHeight = parseInt(dialogRef.current.style.height)

    dialogRef.current.style.transition = isReduced ? 'none' : '0.3s ease'

    if (sheetHeight < 10) return hideBottomSheet()
    if (sheetHeight > 75) return updateSheetHeight(FULL_HEIGHT)

    updateSheetHeight(HALF_HEIGHT)
  }

  const dragStart = (e: MouseEvent | TouchEvent) => {
    if (!dialogRef.current) return

    e.preventDefault()

    if (e.type === 'touchstart' && 'touches' in e) {
      startY.current = e.touches[0].pageY
    } else if ('clientX' in e) {
      startY.current = e.pageY
    }

    startHeight.current = parseInt(dialogRef.current.style.height)
    isDragging.current = true
    dialogRef.current.style.transition = 'none'
  }

  const dragging = (e: MouseEvent | TouchEvent) => {
    if (!dialogRef.current || !isDragging.current) return

    e.preventDefault()

    let pageY
    if (e.type === 'touchmove' && 'touches' in e) {
      pageY = e.touches[0].pageY
    } else if ('clientX' in e) {
      pageY = e.pageY
    }

    const delta = startY.current - (pageY || 0)
    const newHeight = startHeight.current + (delta / window.innerHeight) * 100
    updateSheetHeight(newHeight)
  }

  const onSliderChange = (e: ChangeEvent<HTMLInputElement>) => {
    if (!isDragging.current) {
      const value = parseInt(e.target.value)
      if (value === 1) {
        updateSheetHeight(HALF_HEIGHT)
      } else if (value === 2) {
        updateSheetHeight(FULL_HEIGHT)
      }
    }
  }

  // ü™ù HOOKS

  // Ensures an animation upon closing the component
  useEffect(() => {
    if (!fireDelayedOnClose) return
    setIsOpen(false)
    const timer = setTimeout(() => {
      onClose()
    }, ANIMATION_DURATION)
    return () => clearTimeout(timer)
  }, [fireDelayedOnClose, onClose])

  // Animates the bottom sheet in on mount
  useEffect(() => {
    showBottomSheet()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  const currentSliderValue = snappedHeight === HALF_HEIGHT ? 1 : 2

  return (
    <FullScreenContainer
      open={open}
      onMouseUp={dragStop}
      onMouseMove={dragging}
      onTouchEnd={dragStop}
      onTouchMove={dragging}
    >
      <Overlay data-testid="overlay" onClick={() => hideBottomSheet()}></Overlay>
      <Content
        ref={dialogRef}
        role={role}
        open={open}
        aria-labelledby={ariaLabelledby}
        aria-describedby={ariaDescribedby}
        aria-modal={ariaModal}
        sx={sx}
      >
        <Box sx={{position: 'relative'}}>
          <DraggableRegion
            onMouseDown={dragStart}
            onTouchStart={dragStart}
            onChange={onSliderChange}
            type="range"
            role="slider"
            tabIndex={0}
            min={1}
            max={2}
            value={currentSliderValue}
            aria-label="Draggable dialog height resizer"
            aria-valuemin={1}
            aria-valuemax={2}
            aria-valuenow={currentSliderValue}
            aria-valuetext={`Dialog height ${snappedHeight}% of the screen`}
          />
          <DraggableRegionPill />
          {header}
        </Box>
        {children}
      </Content>
    </FullScreenContainer>
  )
})

// üñåÔ∏è Styles

const FullScreenContainer = styled.div<{open: boolean}>`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  opacity: ${props => (props.open ? 1 : 0)};
  pointerevents: ${props => (props.open ? 'auto' : 'none')};
  flex-direction: column;
  justify-content: flex-end;
  align-items: center;
  transition: ${ANIMATION_DURATION / 3}ms linear;
  @media (prefers-reduced-motion) {
    transition: none;
  }
`
const Overlay = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  z-index: -1;
  width: 100%;
  height: 100%;
  background-color: ${get('colors.primer.canvas.backdrop')};
`

const DraggableRegionPill = styled.div`
  height: 6px;
  width: 70px;
  display: block;
  background-color: ${get('colors.border.muted')};
  border-radius: 3px;
  top: ${get('space.2')};
  position: absolute;
  pointer-events: none;
  left: 50%;
  margin-left: -35px;
`

const DraggableRegion = styled.input`
  appearance: none;
  display: flex;
  justify-content: center;
  border: none;
  position: absolute;
  background: transparent;
  top: 0;
  width: 100%;
  right: 0;
  left: 0;
  bottom: 0;
  padding-top: ${get('space.2')};
  padding-bottom: ${get('space.1')};
  cursor: grab;
  user-select: none;
  min-height: 50px;
  &:hover ~ ${DraggableRegionPill} {
    background-color: ${get('colors.border.default')};
  }
  &:focus {
    outline: none;
    opacity: 0;
  }
  &:focus-visible:not([disabled]) ~ ${DraggableRegionPill} {
    background-color: ${get('colors.accent.emphasis')};
  }
  ::-moz-range-thumb {
    appearance: none;
    visibility: hidden;
  }
  ::-webkit-slider-thumb {
    appearance: none;
  }
`
const Content = styled.div<
  {
    open: boolean
  } & SxProp
>`
  display: flex;
  flex-direction: column;
  background-color: ${get('colors.canvas.default')};
  width: 100vw;
  max-width: 100dvh;
  max-width: 480px;
  border-radius: 12px 12px 0 0;
  position: relative;
  height: 50vh;
  height: 50dvh;
  box-shadow: ${get('shadows.overlay.shadow')};
  max-height: 100vh;
  max-height: 100dvh;
  overflow-x: hidden;
  overflow-y: auto;
  transform: ${props => (props.open ? 'translateY(0%)' : 'translateY(100%)')};
  transition: ${ANIMATION_DURATION}ms ease;
`
