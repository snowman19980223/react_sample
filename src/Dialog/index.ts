export {default, DialogProps, DialogHeaderProps} from './Dialog'
