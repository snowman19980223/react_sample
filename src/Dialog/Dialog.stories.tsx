import React, {useState} from 'react'
import {Meta} from '@storybook/react'

import {BaseStyles, ThemeProvider} from '..'
import {Button} from '../Button'
import Dialog from './Dialog'

export default {
  title: 'Components/Dialog',
  component: Dialog,
  decorators: [
    Story => {
      // Since portal roots are registered globally, we need this line so that each storybook
      // story works in isolation.
      return (
        <ThemeProvider>
          <BaseStyles>
            <Story />
          </BaseStyles>
        </ThemeProvider>
      )
    },
  ],
} as Meta

export const Default = () => {
  const [isOpen, setIsOpen] = useState(false)
  const returnFocusRef = React.useRef(null)

  return (
    <>
      <Button ref={returnFocusRef} onClick={() => setIsOpen(true)}>
        Open
      </Button>
      <Dialog
        returnFocusRef={returnFocusRef}
        isOpen={isOpen}
        onDismiss={() => setIsOpen(false)}
        aria-labelledby="header-id"
      >
        <Dialog.Header id="header-id">Title</Dialog.Header>
        some content
      </Dialog>
    </>
  )
}
