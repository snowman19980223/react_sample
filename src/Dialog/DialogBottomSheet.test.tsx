import React from 'react'
import {fireEvent, render, waitFor} from '@testing-library/react'
import {Dialog} from './Dialog'
import MatchMediaMock from 'jest-matchmedia-mock'
import {behavesAsComponent} from '../utils/testing'
import {FULL_HEIGHT, HALF_HEIGHT} from './DialogBottomSheet'
import {axe} from 'jest-axe'

let matchMedia: MatchMediaMock

describe('Dialog', () => {
  beforeEach(() => {
    matchMedia = new MatchMediaMock()
  })

  afterEach(() => {
    matchMedia.clear()
  })

  behavesAsComponent({
    Component: Dialog,
    options: {skipAs: true, skipSx: true, htmlRenderer: true},
    toRender: () => (
      <Dialog onClose={() => {}} type="bottom-sheet">
        <div>Hello World</div>
      </Dialog>
    ),
  })

  // Arrow keys support in the testing library is currently not implemented and therefore can't be tested
  // https://github.com/testing-library/user-event/issues/871

  it('expands/contracts when using the range slider', async () => {
    const onClose = jest.fn()

    const {getByRole} = render(
      <Dialog onClose={onClose} type="bottom-sheet">
        <div>Hello World</div>
      </Dialog>,
    )

    const slider = getByRole('slider')
    slider.focus()

    expect(slider).toHaveFocus()

    fireEvent.change(slider, {target: {value: 2}})

    const dialog = getByRole('dialog')

    expect(dialog).toHaveStyle(`height: ${FULL_HEIGHT}dvh)`)

    fireEvent.change(slider, {target: {value: 1}})

    expect(dialog).toHaveStyle(`height: ${HALF_HEIGHT}dvh)`)
  })

  it('automatically focuses the footer button when `autoFocus` is true', async () => {
    const {getByRole} = render(
      <Dialog
        onClose={() => {}}
        footerButtons={[{buttonType: 'primary', content: 'Footer button', autoFocus: true}]}
        type={'bottom-sheet'}
      >
        Pay attention to me
      </Dialog>,
    )

    await waitFor(() => expect(getByRole('button', {name: 'Footer button'})).toHaveFocus())
  })

  it('changes the <body> style for `overflow` if it is not set to "hidden"', () => {
    document.body.style.overflow = 'scroll'

    const {container} = render(
      <Dialog onClose={() => {}} type="bottom-sheet">
        Pay attention to me
      </Dialog>,
    )

    expect(container.ownerDocument.body.style.overflow).toBe('hidden')
  })

  it('should have no axe violations', async () => {
    const {container} = render(
      <Dialog type="bottom-sheet" onClose={() => {}}>
        Pay attention to me
      </Dialog>,
    )
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })

  it('opens the bottom sheet on mount', () => {
    const {getByRole} = render(
      <Dialog onClose={() => {}} type="bottom-sheet">
        My dialog content
      </Dialog>,
    )

    expect(getByRole('dialog')).toHaveStyle(`height: ${HALF_HEIGHT}dvh)`)
  })

  it('calls `onClose` when keying "Escape"', () => {
    const onClose = jest.fn()
    const {container} = render(
      <Dialog onClose={onClose} type="bottom-sheet">
        Pay attention to me
      </Dialog>,
    )

    expect(onClose).not.toHaveBeenCalled()

    fireEvent.keyDown(container, {key: 'Escape'})

    expect(onClose).toHaveBeenCalled()
  })

  it('`onClose` is called when clicking close', async () => {
    const onClose = jest.fn()
    const {getByLabelText} = render(
      <Dialog onClose={onClose} type="bottom-sheet">
        <div>Hello World</div>
      </Dialog>,
    )

    expect(onClose).not.toHaveBeenCalled()

    fireEvent.click(getByLabelText('Close'))

    expect(onClose).toHaveBeenCalled()
  })

  it('is dialog and has the aria-model attribute', () => {
    const {getByRole} = render(
      <Dialog onClose={() => {}} type="bottom-sheet">
        My dialog content
      </Dialog>,
    )

    const dialog = getByRole('dialog')
    expect(dialog).toHaveAttribute('aria-modal', 'true')
  })

  it('`onClose` is called when clicking the overlay', async () => {
    const mockOnClose = jest.fn()

    const {getByTestId} = render(
      <Dialog onClose={mockOnClose} type="bottom-sheet">
        My dialog content
      </Dialog>,
    )
    expect(mockOnClose).not.toHaveBeenCalled()
    fireEvent.click(getByTestId('overlay'))
    expect(mockOnClose).toHaveBeenCalled()
  })
})
