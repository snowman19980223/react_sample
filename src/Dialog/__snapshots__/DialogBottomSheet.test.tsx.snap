// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders consistently 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c3 {
  position: relative;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 6px;
  padding-bottom: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c10 {
  z-index: 2;
  right: 0;
  padding-right: 8px;
  position: absolute;
  pointer-events: auto;
}

.c13 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c14 {
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c11 {
  border-radius: 2;
  border: 1px solid;
  border-color: var(--button-default-borderColor-rest,undefined);
  font-family: inherit;
  font-weight: semibold;
  font-size: 14px;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 32px;
  padding: 0 12px;
  gap: 8px;
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
  -webkit-transition: 80ms cubic-bezier(0.65,0,0.35,1);
  transition: 80ms cubic-bezier(0.65,0,0.35,1);
  -webkit-transition-property: color,fill,background-color,border-color;
  transition-property: color,fill,background-color,border-color;
  color: btn.text;
  background-color: btn.bg;
  box-shadow: undefined,undefined;
}

.c11:focus:not(:disabled) {
  box-shadow: none;
  outline: 2px solid;
  outline-offset: -2px;
}

.c11:focus:not(:disabled):not(:focus-visible) {
  outline: solid 1px transparent;
}

.c11:focus-visible:not(:disabled) {
  box-shadow: none;
  outline: 2px solid;
  outline-offset: -2px;
}

.c11[href] {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c11[href]:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11:hover {
  -webkit-transition-duration: 80ms;
  transition-duration: 80ms;
}

.c11:active {
  -webkit-transition: none;
  transition: none;
}

.c11[data-inactive] {
  cursor: auto;
}

.c11:disabled {
  cursor: not-allowed;
  box-shadow: none;
  color: primer.fg.disabled;
}

.c11:disabled [data-component=ButtonCounter] {
  color: inherit;
}

.c11 [data-component=ButtonCounter] {
  font-size: 12px;
}

.c11[data-component=IconButton] {
  display: inline-grid;
  padding: unset;
  place-content: center;
  width: 32px;
  min-width: unset;
}

.c11[data-size="small"] {
  padding: 0 8px;
  height: 28px;
  gap: 4px;
  font-size: 12px;
}

.c11[data-size="small"] [data-component="text"] {
  line-height: calc(20 / 12);
}

.c11[data-size="small"] [data-component=ButtonCounter] {
  font-size: 12px;
}

.c11[data-size="small"] [data-component="buttonContent"] > :not(:last-child) {
  margin-right: 4px;
}

.c11[data-size="small"][data-component=IconButton] {
  width: 28px;
  padding: unset;
}

.c11[data-size="large"] {
  padding: 0 16px;
  height: 40px;
  gap: 8px;
}

.c11[data-size="large"] [data-component="buttonContent"] > :not(:last-child) {
  margin-right: 8px;
}

.c11[data-size="large"][data-component=IconButton] {
  width: 40px;
  padding: unset;
}

.c11[data-block="block"] {
  width: 100%;
}

.c11[data-inactive]:not([disabled]) {
  background-color: var(--button-inactive-bgColor,undefined);
  border-color: var(--button-inactive-bgColor,undefined);
  color: var(--button-inactive-fgColor,undefined);
}

.c11[data-inactive]:not([disabled]):focus-visible {
  box-shadow: none;
}

.c11 [data-component="leadingVisual"] {
  grid-area: leadingVisual;
}

.c11 [data-component="text"] {
  grid-area: text;
  line-height: calc(20/14);
  white-space: nowrap;
}

.c11 [data-component="trailingVisual"] {
  grid-area: trailingVisual;
}

.c11 [data-component="trailingAction"] {
  margin-right: -4px;
}

.c11 [data-component="buttonContent"] {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  display: grid;
  grid-template-areas: "leadingVisual text trailingVisual";
  grid-template-columns: min-content minmax(0,auto) min-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c11 [data-component="buttonContent"] > :not(:last-child) {
  margin-right: 8px;
}

.c11:hover:not([disabled]):not([data-inactive]) {
  background-color: btn.hoverBg;
  border-color: var(--button-default-borderColor-hover,undefined);
}

.c11:active:not([disabled]):not([data-inactive]) {
  background-color: btn.activeBg;
  border-color: var(--button-default-borderColor-active,undefined);
}

.c11[aria-expanded=true] {
  background-color: btn.activeBg;
  border-color: var(--button-default-borderColor-active,undefined);
}

.c11 [data-component="leadingVisual"],
.c11 [data-component="trailingVisual"],
.c11 [data-component="trailingAction"] {
  color: var(--button-color,undefined);
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
  pointerevents: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 100ms linear;
  transition: 100ms linear;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  z-index: -1;
  width: 100%;
  height: 100%;
}

.c5 {
  height: 6px;
  width: 70px;
  display: block;
  border-radius: 3px;
  top: 8px;
  position: absolute;
  pointer-events: none;
  left: 50%;
  margin-left: -35px;
}

.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  position: absolute;
  background: transparent;
  top: 0;
  width: 100%;
  right: 0;
  left: 0;
  bottom: 0;
  padding-top: 8px;
  padding-bottom: 4px;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: 50px;
}

.c4:focus {
  outline: none;
  opacity: 0;
}

.c4::-moz-range-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  visibility: hidden;
}

.c4::-webkit-slider-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100vw;
  max-width: 100dvh;
  max-width: 480px;
  border-radius: 12px 12px 0 0;
  position: relative;
  height: 50vh;
  height: 50dvh;
  max-height: 100vh;
  max-height: 100dvh;
  overflow-x: hidden;
  overflow-y: auto;
  -webkit-transform: translateY(0%);
  -ms-transform: translateY(0%);
  transform: translateY(0%);
  -webkit-transition: 300ms ease;
  transition: 300ms ease;
}

.c6 {
  box-shadow: 0 1px 0;
  padding: 8px;
  z-index: 1;
  pointer-events: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
}

.c9 {
  font-size: 14px;
  font-weight: 600;
  margin: 0;
  width: calc(100% - 24px);
}

.c15 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: auto;
  padding: 16px;
}

.c12 {
  border-radius: 4px;
  background: transparent;
  border: 0;
  vertical-align: middle;
  padding: 8px;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  line-height: normal;
  box-shadow: none;
}

@media (forced-colors:active) {
  .c11:focus {
    outline: solid 1px transparent;
  }
}

@media (prefers-reduced-motion) {

}

@media (prefers-reduced-motion) {
  .c0 {
    -webkit-transition: none;
    transition: none;
  }
}

<body
    style="overflow: hidden;"
  >
    <div />
    <div
      id="__primerPortalRoot__"
      style="position: absolute; top: 0px; left: 0px;"
    >
      <div
        style="position: relative; z-index: 1;"
      >
        <div
          class="c0"
          open=""
        >
          <div
            class="c1"
            data-testid="overlay"
          />
          <div
            aria-describedby=":r1:"
            aria-labelledby=":r0:"
            aria-modal="true"
            class="c2"
            data-focus-trap="active"
            open=""
            role="dialog"
          >
            <span
              aria-hidden="true"
              class="sentinel"
              tabindex="0"
            />
            <div
              class="c3"
            >
              <input
                aria-label="Draggable dialog height resizer"
                aria-valuemax="2"
                aria-valuemin="1"
                aria-valuenow="1"
                aria-valuetext="Dialog height 50% of the screen"
                class="c4"
                max="2"
                min="1"
                role="slider"
                tabindex="0"
                type="range"
                value="1"
              />
              <div
                class="c5"
              />
              <div
                class="c6"
              >
                <div
                  class="c7"
                  display="flex"
                >
                  <div
                    class="c8"
                    display="flex"
                  >
                    <h1
                      class="c9"
                      id=":r0:"
                    >
                      Dialog
                    </h1>
                  </div>
                  <div
                    class="c10"
                  >
                    <button
                      aria-label="Close"
                      class="c11 c12 focus-visible"
                      data-focus-visible-added=""
                      data-no-visuals="true"
                      type="button"
                    >
                      <span
                        class="c13"
                        data-component="buttonContent"
                      >
                        <span
                          data-component="text"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            fill="currentColor"
                            focusable="false"
                            height="16"
                            role="img"
                            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
                            viewBox="0 0 16 16"
                            width="16"
                          >
                            <path
                              d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z"
                            />
                          </svg>
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="c14 DialogOverflowWrapper"
            >
              <div
                class="c15"
              >
                <div>
                  Hello World
                </div>
              </div>
            </div>
            <span
              aria-hidden="true"
              class="sentinel"
              tabindex="0"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": @media (forced-colors:active) {

}

@media (prefers-reduced-motion) {

}

@media (prefers-reduced-motion) {

}

<div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
